sdk/couchbase-core/src/httpx/base.rs

```diff
@@ -0,0 +1,103 @@
+use async_trait::async_trait;
+use bytes::Bytes;
+use futures::{Stream, StreamExt};
+use http::{StatusCode};
+use serde::de::DeserializeOwned;
+use typed_builder::TypedBuilder;
+use crate::httpx::error::ErrorKind::Generic;
+use crate::httpx::error::{Result as HttpxResult};
+
+#[derive(Debug, TypedBuilder)]
+#[builder(field_defaults(default, setter(into)))]
+#[non_exhaustive]
+pub struct Request {
```

Comment:
Sorry, I missed this before. Most of these fields in these various structs should probably be Option<> if they are actually optional.